File: lib/core/checkout/interfaces/checkout_service.dart
import 'package:sertton/core/checkout/dtos/installment_dto.dart';
import 'package:sertton/core/checkout/dtos/payment_dto.dart';
import 'package:sertton/core/checkout/dtos/order_dto.dart';
import 'package:sertton/core/global/responses/rest_response.dart';

abstract class CheckoutService {
  Future<RestResponse<String>> fetchCheckoutLink(
    List<String> skuTokens,
    List<int> quantities,
  );
  Future<RestResponse<List<OrderDto>>> fetchOrdersByCustomer(
    String customerDocument,
  );
  Future<RestResponse<List<PaymentDto>>> fetchPayments();
  Future<RestResponse<List<InstallmentDto>>> fetchInstallments(
    String paymentId,
    String productId,
    double productPrice,
  );
}

File: sertton-react-native-app/src/core/checkout/interfaces/checkout-service.ts
import type { Installment, Order, Payment } from "@/core/checkout/entities"
import type { RestResponse } from "@/core/shared/responses/rest-response"

export interface CheckoutService {
  fetchCheckoutLink(
    skuTokens: readonly string[],
    quantities: readonly number[],
  ): Promise<RestResponse<string>>
  fetchInstallments(
    paymentId: string,
    productId: string,
    productPrice: number,
  ): Promise<RestResponse<readonly Installment[]>>
  fetchOrdersByCustomer(customerDocument: string): Promise<RestResponse<readonly Order[]>>
  fetchPayments(): Promise<RestResponse<readonly Payment[]>>
}

export type ICheckoutService = CheckoutService
